### PR TITLE
s3: Pick a pool with >85% if all other pools are in suspended state

### PR DESCRIPTION
**Pull request by vadmeste**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/erasure-server-pool.go [hunk 1]
```diff
@@ -299,7 +301,7 @@ func (p serverPoolsAvailableSpace) FilterMaxUsed(max int) {
 
 	// Remove entries that are above.
 	for i, z := range p {
-		if z.MaxUsedPct < max {
+		if z.Valid && z.MaxUsedPct < max {
```
Comment (harshavardhana) [inline cmd/erasure-server-pool.go]:
```suggestion
		if z.MaxUsedPct > 0 && z.MaxUsedPct < max {
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/erasure-server-pool.go [hunk 2]
```diff
@@ -265,6 +265,8 @@ type poolAvailableSpace struct {
 	Index      int
 	Available  uint64
 	MaxUsedPct int // Used disk percentage of most filled disk, rounded down.
+
+	Valid bool
```
Comment (klauspost) [inline cmd/erasure-server-pool.go]: "Valid" without a bit more context isn't very obvious. A comment would be great.

File: cmd/erasure-server-pool.go (hunk 1 above)
Comment (klauspost) [reply cmd/erasure-server-pool.go]: @harshavardhana It seems to me this would break on new installs. Maybe @vadmeste can explain the original code.

Comment (vadmeste) [reply cmd/erasure-server-pool.go]: yeah it is risky for me to ignore pools with 0% of usage, such as (0.11%)

File: cmd/erasure-server-pool.go (hunk 2 above)
Comment (vadmeste) [reply cmd/erasure-server-pool.go]: > "Valid" without a bit more context isn't very obvious. A comment would be great.

I removed it. I am using Available now.

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**